sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -207,13 +256,35 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
       val distinctAggChildren = distinctAggGroups.keySet.flatten.toSeq.distinct
       val distinctAggChildAttrMap = distinctAggChildren.map(expressionAttributePair)
       val distinctAggChildAttrs = distinctAggChildAttrMap.map(_._2)
+      // Setup all the filters in distinct aggregate.
+      val distinctAggExprs = aggExpressions
+        .filter(e => e.isDistinct && e.children.exists(!_.foldable))
+      val distinctAggFilterAttrMap = distinctAggExprs.collect {
```

Comment:
nit: `val (distinctAggFilters, distinctAggFilterAttrs, maxCond) = distinctAggExprs.collect(...).unzip3`

Reply:
But I want
```
val distinctAggFilterAttrLookup = distinctAggFilterAttrMap.map { tuple3 =>
        tuple3._1 -> tuple3._3.toAttribute
      }.toMap
```

Reply:
this is `distinctAggFilters.zip(maxCond.map(_.toAttribute)).toMap`

Reply:
OK